front/lib/upsert_document.ts

```diff
@@ -66,7 +79,12 @@ export async function enqueueUpsertDocument({
         upsertQueueId,
         workspaceId: upsertDocument.workspaceId,
         dataSourceName: upsertDocument.dataSourceName,
-        documentId: upsertDocument.documentId,
+        documentId: EnqueueUpsertDocument.is(upsertDocument)
```

Comment:
You can then remove the `.is()` once you fix the type.

Reply:
This branching will disappear with 2 workflows

Reply:
oh actually maybe it won't :smiling_imp: